src/renderer/components/ft-subscribe-button/ft-subscribe-button.js

```diff
@@ -0,0 +1,141 @@
+      return this.activeProfile.subscriptions.find((channel) => {
+        return channel.id === this.channelId
+      }) ?? null
+    },
+
+    isSubscribed: function () {
+      return this.subscriptionInfo !== null
+    },
+
+    subscribedText: function () {
+      let subscribedValue = (this.isSubscribed ? this.$t('Channel.Unsubscribe') : this.$t('Channel.Subscribe')).toUpperCase()
+      if (this.subscriptionCountText !== '') {
+        subscribedValue += ' ' + this.subscriptionCountText
+      }
+      return subscribedValue
+    }
+
+  },
+  methods: {
+    handleSubscription: function () {
+      if (this.channelId === '') {
+        return
+      }
+
+      const currentProfile = JSON.parse(JSON.stringify(this.activeProfile))
+      const primaryProfile = JSON.parse(JSON.stringify(this.profileList[0]))
+
+      if (this.isSubscribed) {
+        currentProfile.subscriptions = currentProfile.subscriptions.filter((channel) => {
+          return channel.id !== this.channelId
+        })
+
+        this.updateProfile(currentProfile)
+        showToast(this.$t('Channel.Channel has been removed from your subscriptions'))
+
+        if (this.activeProfile._id === MAIN_PROFILE_ID) {
+          // Check if a subscription exists in a different profile.
+          // Remove from there as well.
+          let duplicateSubscriptions = 0
+
+          this.profileList.forEach((profile) => {
+            if (profile._id === MAIN_PROFILE_ID) {
+              return
+            }
+            duplicateSubscriptions += this.unsubscribe(profile, this.channelId)
+          })
+
+          if (duplicateSubscriptions > 0) {
+            const message = this.$t('Channel.Removed subscription from {count} other channel(s)', { count: duplicateSubscriptions })
+            showToast(message)
+          }
+        }
+      } else {
+        const subscription = {
+          id: this.channelId,
+          name: this.channelName,
+          thumbnail: this.channelThumbnail
+        }
+        currentProfile.subscriptions.push(subscription)
+
+        this.updateProfile(currentProfile)
+        showToast(this.$t('Channel.Added channel to your subscriptions'))
+
+        if (this.activeProfile._id !== MAIN_PROFILE_ID) {
+          const index = primaryProfile.subscriptions.findIndex((channel) => {
+            return channel.id === this.channelId
+          })
+
+          if (index === -1) {
+            primaryProfile.subscriptions.push(subscription)
+            this.updateProfile(primaryProfile)
+          }
+        }
+      }
+    },
+
+    unsubscribe: function(profile, channelId) {
+      const parsedProfile = JSON.parse(JSON.stringify(profile))
+      const index = parsedProfile.subscriptions.findIndex((channel) => {
+        return channel.id === channelId
+      })
+
+      if (index !== -1) {
+        parsedProfile.subscriptions = parsedProfile.subscriptions.filter((x) => {
+          return x.id !== channelId
+        })
```

Comment:
Untested suggestion 🙈
As you already have the index from findIndex, this filter could probably be replaced with `.splice(index, 1)` to remove that item from the array without needing to do the filter (iterates over the entire array again and creates a new one). Should be more performant but I haven't tested if it works.
```suggestion
        parsedProfile.subscriptions.splice(index, 1)
```

Reply:
I think the filter may have been there to fix issues with being subscribed to the same channel multiple times that used to happen